### ISSUE TITLE
Round 1: Strategic upstream OSS support by reviewing non-maintainer PRs

### ISSUE THREAD

**Issue by yuvipanda** [2i2c-org/infrastructure#5058]
This issue tracks one round of doing strategic upstream work by reviewing PRs from non-maintainers in the communities we care about. The primary goals here are:

1. Shake out our process for integrating this into our regular planning, so everyone in the team can participate
2. Provide a place to track what work we are actually doing
3. Build capacity within our team for doing code review *consistently* and *equitably*.

Round 1 should continue until the end of January, where we can evaluate the results.

## Process

WIP process that will be modified as we go.

1. We create a subissue of this issue for each PR we want to review
2. Created subissue gets tracked as a task in P&S sprint planning

**Comment by agoose77** [comment]
@sgibson91 and I took a look at jupyterhub/nbgitpuller#359 (#5059). 

Overall the PR is indeed simple, and looks correct. We clarified the issue cause, and identified that the suggested PR was a suitable fix. We made a suggestion about future improvements to this work.

We noticed that there is a range of experience in the team. We reflected that this was therefore a good opportunity not only to contribute to OSS, but also to become more familiar with keystone projects in the ecosystem, and in particular, to undertake some peer mentoring in order to narrow the range of experience and skills.

On that note, @agoose77 originally suggested a 30m meeting. This was overly optimistic, and didn't facilitate a good opportunity for peer mentoring.

From @agoose77, thanks @sgibson91 for pairing!

**Comment by GeorgianaElena** [comment]
@agoose77 and I paired on https://github.com/2i2c-org/infrastructure/issues/5183 today.

We started by discussing the different approaches we each had in mind for approaching the task, which was very interesting to acknowledge, celebrate and learn from.

We then spent some time dodging docker caching in order to be able to reproduce the warning multiple times.

After getting both of our environments ready for action we realized the PR was a bit more complex than expected because we both got lured into a mindset of wanting to optimize even more. Use used all the time we had set aside for this task (1h) in order for the both of us to become confident with merging the changes and ended up opening a new follow-up issue with our exploratory findings.

All in one it was a very enjoyable pairing session, thank you @agoose77 🚀

**Comment by sgibson91** [comment]
Follow-up to https://github.com/2i2c-org/infrastructure/issues/5323

Making a release upstream was a pretty seamless experience tbh since it is very automated through the combination of `github-activity`, `tbump`, and CI. I think more guidance could be given around deciding version numbers. I also wasn't sure about merging the changelog PR and opted to bias towards action and self-merge.

**Comment by jnywong** [comment]
Follow-up to #5251:

@GeorgianaElena and I paired up on the nbgitpuller 1.2.2 release. We followed the guidance in the repo's [RELEASE.md](https://github.com/jupyterhub/nbgitpuller/blob/main/RELEASE.md) and it went smoothly! Georgiana explained a lot as she went through the pairing session with me, making references to the JupyterHub Team Compass and explaining the [python package template](https://github.com/jupyterhub/jupyterhub-python-repo-template).

Great knowlege-sharing achieved in this pairing session, thank you!

**Comment by agoose77** [comment]
Follow-up to #5465 

@sgibson91 and I paired on reviewing this PR. We followed the prompts in @yuvipanda's refinement, and set-about checking the text-diff for obvious mistakes, as well as attempting to test the CLI.

This project does not mention the test-suite in the README (in place of contributing), and it is clear that we needed to spin up a local hub if we wanted to validate whether the PR breaks the existing culler behaviour. Nevertheless, the CI is running the test suite, so we are comfortable that having not-tested this PR directly is not an impediment to merging it.

@sgibson91 gave me some good feedback before we started about improvements to the pairing process, which I found helpful.

**Comment by jnywong** [comment]
FYI

> Round 1 should continue until the end of January, where we can evaluate the results.

😄

**Comment by jnywong** [comment]
Sarah and I tackled: https://github.com/2i2c-org/infrastructure/issues/5753.

This one was straightforward and we teamed up pretty well: I was comfortable with validating the issue with xsrf and Sarah was happy with that and cut a release.

**Comment by jnywong** [comment]
@agoose77 and I looked at https://github.com/2i2c-org/infrastructure/issues/5854

This took a couple of sync sessions, since there were 3 PRs to review with a bit of homework for us in between. 

We landed on some solid outcomes (see [comment](https://github.com/2i2c-org/infrastructure/issues/5854)) but lack the merge rights to follow through and tie this off in the current sprint.

**Comment by yuvipanda** [comment]
I was away for a while and that stopped the OSS contributions. I want to build systems that sustain it while I'm not here. In the meantime, I'll start doing this again.

**Comment by yuvipanda** [comment]
I'm working separately with some JupyterHub folks to systematize that board. It's *fully* automated, no human action required (maintained via https://github.com/yuvipanda/pr-triage-syncer) but let's not focus on that too much. Let's focus on how we as 2i2c can use it!
